src/Microsoft.AspNet.Mvc.Core/OutputFormatterOptions.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNet.Mvc.Core
+{
+    public class OutputFormatterOptions
+    {
+        private Dictionary<string, MediaTypeHeaderValue> map = new Dictionary<string, MediaTypeHeaderValue>();
+
+        public void AddFormatMapping(string format, MediaTypeHeaderValue contentType)
+        {
+            if (!string.IsNullOrEmpty(format) && contentType != null)
```

Comment:
Why do any of this? Typically you want to throw if input is invalid, not accept it. If something invalid gets in here and we ignore it, the user will want to know!